src/Test/Utilities/Portable/Assert/AssertEx.cs

```diff
@@ -265,6 +265,95 @@ private static bool SequenceEqual<T>(IEnumerable<T> expected, IEnumerable<T> act
             return true;
         }
 
+        public static void SetEqual(IEnumerable<string> expected, IEnumerable<string> actual, IEqualityComparer<string> comparer = null, string message = null, string itemSeparator = "\r\n", Func<string, string> itemInspector = null)
+        {
+            var indexes = new Dictionary<string, int>();
+            int counter = 0;
+            foreach (var expectedItem in expected)
+            {
+                indexes.Add(expectedItem, counter++);
+            }
+
+            SetEqual<string>(expected, actual.OrderBy(e => getIndex(e)), comparer, message, itemSeparator, itemInspector);
+
+            int getIndex(string item)
+            {
+                // exact match to expected items
+                if (indexes.TryGetValue(item, out var index))
+                {
+                    return index;
+                }
+
+                // closest match to expected items
+                int closestDistance = int.MaxValue;
+                string closestItem = null;
+                foreach (var expectedItem in indexes.Keys)
+                {
+                    var distance = leventshtein(item, expectedItem);
+                    if (distance < closestDistance)
+                    {
+                        closestDistance = distance;
+                        closestItem = expectedItem;
+                    }
+                }
+
+                if (closestItem != null)
+                {
+                    _ = indexes.TryGetValue(closestItem, out index);
+                    return index;
+                }
+
+                return -1;
+            }
+
+            // Adapted from Toub's https://blogs.msdn.microsoft.com/toub/2006/05/05/generic-levenshtein-edit-distance-with-c/
+            int leventshtein(string first, string second)
```

Comment:
I know this is really pedantic, but Levenshtein only has one "t".

Reply:
Thanks. Fixed the typo